### ISSUE TITLE
Folding HereStrings

### ISSUE THREAD

**Issue by mpearon** [PowerShell/vscode-powershell#704]
### System Details

- Operating system name and version:  **Windows 10 v1607 Build 14939.1066**
- VS Code version:  **1.11.2**
- PowerShell extension version:  **0.12.2**
- Output from `$PSVersionTable`:
                                                                                                            
**PSVersion                      5.1.14393.1066**
**PSEdition                      Desktop**
**PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}**
**BuildVersion                   10.0.14393.1066**
**CLRVersion                     4.0.30319.42000**
**WSManStackVersion              3.0**
**PSRemotingProtocolVersion      2.3**
**SerializationVersion           1.1.0.1**

### Issue Description

I'm sure this has been mentioned before, but I can't seem to find any open tickets. 
Any time HereStrings are used throughout my scripts, 1) they do not fold and 2) they break folding for parent elements (Functions or Regions in my case).

Here is a gif of the behavior, and I've included the exact code for you to play with below.
![brokenfolding](https://cloud.githubusercontent.com/assets/7157790/25558598/a4b8bb78-2cef-11e7-82f0-712b9510ef94.gif)

```
Function foldingDemo{
    #Region > Proper Fold >
        $testString = 'This region folds'
    #Region < Proper Fold <

    #Region > Broken Fold >
        $testHereString = @'
- The HereString does not offer folding
- This region does not fold properly
- The folding for the function breaks at this point
'@
    #Region < Broken Fold <
}
```

**Comment by rkeithhill** [comment]
Duplicate of #525.  I encourage you to complain to the VSCode team via this issue: https://github.com/Microsoft/vscode/issues/3422

**Comment by mpearon** [comment]
Thank you @rkeithhill.  I've added my name and demo to the list!

**Comment by rkeithhill** [comment]
@mpearon Thanks!  Every bit of feedback to the VSCode team helps (I hope).  :-)

**Comment by mpearon** [comment]
I just want to make note that the language-aware folding was implemented in PowerShell 1.8.0/1!
Great job, everyone!
